savlatkhon-eshonov-1/task-4.ts

```diff
@@ -0,0 +1,186 @@
+interface Product {
+  name: string;
+  price: number;
+  stock: number
+}
+
+interface Customer {
+  name:string;
+  email:string;
+  address:string;
+}
+
+interface ShippingService {
+  shipOrder(order: Order): void;
+}
+
+class Product {
+  constructor(
+    public name:string,
+    public price:number,
+    public stock:number = 0
+  ) {}
+
+  addStock(quantity:number):string {
+    this.stock += quantity;
+    return 'Succesfully added!'
+  }
+
+  reduceStock(quantity:number):string {
+    if (this.stock < quantity) {
+      return 'There are no so much products in stock!'
+    }
+    this.stock -= quantity;
+    return 'Succesfully reduced!'
+  }
+}
+
+class Customer {
+  public cart:Product[] = [];
+  constructor(
+    public name:string,
```

Comment:
make them private and add two getter functions 
`  getCart() {
    return this.cart;
  }

  getAddress() {
    return this.address;
  }`